product/views.py

```diff
@@ -7,11 +7,28 @@
 from product.models import (
   Product, 
   ProductReview, 
-  ReviewLike
+  ReviewLike,
+  Category
 )
 
 DISCOUNT_PROUDCTS_COUNT = 5
 
+class CategoryView(View):
+    def get(self, request):
+        category_list = [{
+            'id': category.id,
+            'name': category.name,
+            'sub_category': [{
+                'id': sub_category.id,
+                'name': sub_category.name,
+                'detail_category' : [{
+                    'id': detail_category.id,
+                    'name': detail_category.name
+                } for detail_category in sub_category.detailcategory_set.all()]
+            } for sub_category in category.subcategory_set.all()]
+            } for category in Category.objects.all().order_by('id')]
+        return JsonResponse({'categories': category_list}, status=200)
```

Comment:
```suggestion
                    'name': detail_category.name
                } for detail_category in sub_category.detailcategory_set.all()]
            } for sub_category in category.subcategory_set.all()]
        } for category in Category.objects.all().order_by('id')]
        return JsonResponse({'categories': category_list}, status=200)
```
아이러니하게...굉장히 복잡하면서도 간결하네요 ㅎㅎ 👍
리스트 들 사이의 포함관계를 시각적으로 명확하게 하기 위해서 괄호를 한 단계 당기시면 좋겠네요!

Reply:
복잡하면서 간결하다니..ㅋㅋㅋㅋㅋㅋ 인덴트 수정했습니다~!